### ISSUE TITLE
Android: Keyboard navigation does not work after updating to 3.3

### ISSUE THREAD

**Issue by APopatanasov** [xamarin/Xamarin.Forms#4561]
### Description
After updating to 3.3 the keyboard navigation that I have in my native Android View which is rendered in Xamarin Forms no longer works. 

The FocusSearch method that is overridden in the VisualElementRenderer seems to causes the issue. It is expected to have more complex native Android controls that could be rendered in Xamarin Forms and breaking the keyboard navigation (respecting only the TabIndex) is not correct. 

### Steps to Reproduce

1. Run the attached project.
2. Focus the first EditText.
3. Try to navigate with the keyboard to the next EditText.

### Expected Behavior
You should be able to see a Button in the keyboard through which you can perform the navigation.

### Actual Behavior
Can no longer navigate using the keyboard.
### Basic Information

- Version with issue: 3.3.0
- Last known good version:3.2.0
- IDE: Visual Studio 2017
- Platform Target Frameworks: 
  - Android: 8.1

### Reproduction Link

[TabIndexBug.zip](https://github.com/xamarin/Xamarin.Forms/files/2623632/TabIndexBug.zip)

**Comment by LanceMcCarthy** [comment]
@samhouts Is this scheduled for any pre4 releases? The latest I see is pre3 and none are new enough to contain the fix.

**Comment by samhouts** [comment]
@LanceMcCarthy You'll find it in the 3.5.0 stable that was released today. Thanks!

**Comment by APopatanasov** [comment]
@samhouts @paymicro The issue should be reopened. I have tested the released 3.5 version and the issues is still observed.

I have checked the fix (https://github.com/xamarin/Xamarin.Forms/pull/4612) and could not agree with it. 

In the provided sample project (attached above) you will notice a Xamarin Forms control that renders a custom native one. The native control is a simple LinearLayout with two **EditText** controls. Each of the controls has **InputTypes** set to **NumberFlagDecimal**. The code that is left in the sample (with NextFocusForwardId) has nothing to do with the issue and the made fix (https://github.com/xamarin/Xamarin.Forms/pull/4612) is based on that irrelevant code. 

The main purpose of the sample was to show the **InputTypes** property which is of great importance for the Android keyboard - not the NextFocusForwardId. Setting the **InputTypes** to the mentioned value for example will visualize **Next** as an action button:
https://developer.android.com/training/keyboard-input/style

With version 3.2 if the first EditText is focused when the next button is pressed the focus will be moved to the next EditText. With 3.3 the FocusSearch method in the VisualElementRenderer is overridden and basically the base logic is scratched. Not respecting the keyboard behavior that comes from the native Android world is major issue for me.

**With version 3.2:**
![expected](https://user-images.githubusercontent.com/9336157/52400194-1d4db080-2ac7-11e9-9515-8762ff2b20f7.gif)
**With version 3.5:**
![unexpected](https://user-images.githubusercontent.com/9336157/52400197-1de64700-2ac7-11e9-89a2-d9846f48fcc6.gif)

**Comment by paymicro** [comment]
@APopatanasov to solve the above issue you can set `NextFocusDownId` for controls inside you custom control.
```C#
editText1.NextFocusForwardId = editText1.NextFocusDownId = editText2.Id;
editText2.NextFocusForwardId = editText2.NextFocusUpId = editText1.Id;
```

**Comment by APopatanasov** [comment]
@paymicro The cause of reported issue is not connected with the NextFocusForwardId. You can check the attached sample with version 3.2 demonstrating the expected behavior and after upgrading to 3.3 the issue occurs. This behavior comes from the native Android framework when **InputTypes** is set to **NumberFlagDecimal** and it is not connected with the NextFocusForwardId or NextFocusDownId.

Applying such a workaround (using NextFocusForwardId or NextFocusDownId) in our control is hard and not something we would like to do, considering that we threat this as a regression issue. Furthermore, we have scenarios where an exception is thrown due to fact that the native focus behavior is not executed accurately after the changes made with 3.3.

[TabIndexBug.zip](https://github.com/xamarin/Xamarin.Forms/files/2840635/TabIndexBug.zip)
